storage/src/test/java/org/apache/kafka/server/log/remote/storage/LocalTieredStorage.java

```diff
@@ -0,0 +1,578 @@
+ * </p>
+ * <p>
+ * The local tiered storage keeps a simple structure of directories mimicking that of Apache Kafka.
+ * <p>
+ * The name of each of the files under the scope of a log segment (the log file, its indexes, etc.)
+ * follows the structure UuidBase64-FileType.
+ * <p>
+ * Given the root directory of the storage, segments and associated files are organized as represented below.
+ * </p>
+ * <code>
+ * / storage-directory  / LWgrMmVrT0a__7a4SasuPA-0-topic / bCqX9U--S-6U8XUM9II25Q.log
+ * .                                                     . bCqX9U--S-6U8XUM9II25Q.index
+ * .                                                     . bCqX9U--S-6U8XUM9II25Q.timeindex
+ * .                                                     . h956soEzTzi9a-NOQ-DvKA.log
+ * .                                                     . h956soEzTzi9a-NOQ-DvKA.index
+ * .                                                     . h956soEzTzi9a-NOQ-DvKA.timeindex
+ * .
+ * / LWgrMmVrT0a__7a4SasuPA-1-topic / o8CQPT86QQmbFmi3xRmiHA.log
+ * .                                . o8CQPT86QQmbFmi3xRmiHA.index
+ * .                                . o8CQPT86QQmbFmi3xRmiHA.timeindex
+ * .
+ * / DRagLm_PS9Wl8fz1X43zVg-3-btopic / jvj3vhliTGeU90sIosmp_g.log
+ * .                                 . jvj3vhliTGeU90sIosmp_g.index
+ * .                                 . jvj3vhliTGeU90sIosmp_g.timeindex
+ * </code>
+ */
+public final class LocalTieredStorage implements RemoteStorageManager {
+
+    public static final String STORAGE_CONFIG_PREFIX = "remote.log.storage.local.";
+
+    /**
+     * The root directory of this storage.
+     */
+    public static final String STORAGE_DIR_PROP = "dir";
+
+    /**
+     * Delete all files and directories from this storage on close, substantially removing it
+     * entirely from the file system.
+     */
+    public static final String DELETE_ON_CLOSE_PROP = "delete.on.close";
+
+    /**
+     * The implementation of the transfer of the data of the canonical segment and index files to
+     * this storage. The only reason the "transferer" abstraction exists is to be able to simulate
+     * file copy errors and exercise the associated failure modes.
+     */
+    public static final String TRANSFERER_CLASS_PROP = "transferer";
+
+    /**
+     * Whether the deleteLogSegment() implemented by this storage should actually delete data or behave
+     * as a no-operation. This allows to simulate non-strongly consistent storage systems which do not
+     * guarantee visibility of a successful delete for subsequent read or list operations.
+     */
+    public static final String ENABLE_DELETE_API_PROP = "delete.enable";
+
+    /**
+     * The ID of the broker which owns this instance of {@link LocalTieredStorage}.
+     */
+    public static final String BROKER_ID = "broker.id";
+
+    private static final String ROOT_STORAGES_DIR_NAME = "kafka-tiered-storage";
+
+    private volatile File storageDirectory;
```

Comment:
any reason to not using Path instead of File?

Reply:
No strong preference, to avoid wrapping the IOException as RuntimeException. We can refactor this in the subsequent PR if required.